src/realm/sync/noinst/client_reset.cpp

```diff
@@ -467,199 +579,111 @@ void client_reset::transfer_group(const Transaction& group_src, Transaction& gro
 
     // Diff all the values and update if needed.
     for (auto table_key : group_src.get_table_keys()) {
+        if (!group_src.table_is_public(table_key))
+            continue;
         ConstTableRef table_src = group_src.get_table(table_key);
         StringData table_name = table_src->get_name();
-        if (!table_name.begins_with("class"))
-            continue;
         TableRef table_dst = group_dst.get_table(table_name);
         REALM_ASSERT(table_src->size() == table_dst->size());
         REALM_ASSERT(table_src->get_column_count() == table_dst->get_column_count());
+        auto pk_col = table_src->get_primary_key_column();
+        REALM_ASSERT(table_src->get_primary_key_column());
+        logger.debug("Updating values for table '%1', number of rows = %2, "
+                     "number of columns = %3, primary_key_col = %4, "
+                     "primary_key_type = %5",
+                     table_name, table_src->size(), table_src->get_column_count(), pk_col.get_index().val,
+                     pk_col.get_type());
 
-        if (auto pk_col = table_src->get_primary_key_column()) {
-            logger.debug("Updating values for table '%1', number of rows = %2, "
-                         "number of columns = %3, primary_key_col = %4, "
-                         "primary_key_type = %5",
-                         table_name, table_src->size(), table_src->get_column_count(), pk_col.get_index().val,
-                         pk_col.get_type());
-        }
-        else {
-            logger.debug("Updating values for table '%1', number of rows = %2, number of columns = %3", table_name,
-                         table_src->size(), table_src->get_column_count());
-        }
+        std::vector<InterRealmValueConverter> columns_cache;
 
+        for (ColKey col_key_src : table_src->get_column_keys()) {
+            if (col_key_src == pk_col)
+                continue;
+            StringData col_name = table_src->get_column_name(col_key_src);
+            ColKey col_key_dst = table_dst->get_column_key(col_name);
+            REALM_ASSERT(col_key_dst);
+            columns_cache.emplace_back(InterRealmValueConverter(table_src, col_key_src, table_dst, col_key_dst));
```

Comment:
I am not sure what is actually cached. Is it just the "opposite tables"? Does it improve the performance much?

Reply:
It saves the column lookup by name in the destination table. It also saves on not having to look up the opposite tables for every linked column. Otherwise those lookups would happen on every object in the table. This improves performance by ~10-20 percent according to the benchmarks.